test/PoolManager.t.sol

```diff
@@ -641,39 +641,18 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
             key, IPoolManager.SwapParams(true, 10000, SQRT_RATIO_1_2), PoolSwapTest.TestSettings(true, true), ZERO_BYTES
         );
 
-        assertEq(manager.protocolFeesAccrued(currency0), expectedFees);
-        assertEq(manager.protocolFeesAccrued(currency1), 0);
+        assertEq(manager.balanceOf(address(feeController), currency0), expectedFees);
+        assertEq(manager.balanceOf(address(feeController), currency1), 0);
         assertEq(currency0.balanceOf(address(1)), 0);
+        vm.prank(address(feeController));
         snapStart("erc20 collect protocol fees");
-        manager.collectProtocolFees(address(1), currency0, expectedFees);
-        snapEnd();
-        assertEq(currency0.balanceOf(address(1)), expectedFees);
-        assertEq(manager.protocolFeesAccrued(currency0), 0);
-    }
-
-    function test_collectProtocolFees_ERC20_returnsAllFeesIf0IsProvidedAsParameter() public {
-        uint24 protocolFee = 260; // 0001 00 00 0100
-        uint256 expectedFees = 7;
-
-        feeController.setSwapFeeForPool(key.toId(), uint16(protocolFee));
-        manager.setProtocolFees(key);
-
-        (Pool.Slot0 memory slot0,,,) = manager.pools(key.toId());
-        assertEq(slot0.protocolFees, protocolFee << 12);
-
-        swapRouter.swap(
-            key, IPoolManager.SwapParams(true, 10000, SQRT_RATIO_1_2), PoolSwapTest.TestSettings(true, true), ZERO_BYTES
-        );
-
-        assertEq(manager.protocolFeesAccrued(currency0), expectedFees);
-        assertEq(manager.protocolFeesAccrued(currency1), 0);
-        assertEq(currency0.balanceOf(address(1)), 0);
         manager.collectProtocolFees(address(1), currency0, 0);
+        snapEnd();
         assertEq(currency0.balanceOf(address(1)), expectedFees);
-        assertEq(manager.protocolFeesAccrued(currency0), 0);
+        assertEq(manager.balanceOf(address(feeController), currency0), 0);
     }
 
-    function test_collectProtocolFees_nativeToken_allowsOwnerToAccumulateFees_gas() public {
+    function test_collectProtocolFees_nativeToken_returnsAllFeesIf0IsProvidedAsParameter_gas() public {
```

Comment:
Would love to see tests for:

- Collecting more fees than they have accrued, should revert
- Invalid caller (although i think we should be removing that, as stated in my previous comment)
- Collecting an amount of fees that is < whole balance
- Collecting a fee where 0 have been accrued (so passing in 0, and the balance is 0)

This could actually be a perfect use for a simple fuzz test:

- argument of `uint256`
- if == 0 expect they claim their whole balance
- if <= their balance expect they claim the amount specified
- if > their balance expect revert

Reply:
lol sorry im a super picky test reviewer as of this week 😆

Reply:
Lol all good yeah I think there def needs some more coverage here. I'll add that fuzz test!